lib/report_portal/cucumber/report.rb

```diff
@@ -183,24 +185,53 @@ def start_feature_with_parentage(feature, desired_time)
               tags = []
               type = :SUITE
             else
+              # TODO: Consider adding feature description and comments.
               name = "#{feature.keyword}: #{feature.name}"
-              description = feature.file # TODO: consider adding feature description and comments
+              description = feature.file
               tags = feature.tags.map(&:name)
               type = :TEST
             end
-            # TODO: multithreading # Parallel formatter always executes scenarios inside the same feature in the same process
-            if parallel? &&
-                index < path_components.size - 1 && # is folder?
-                (id_of_created_item = ReportPortal.item_id_of(name, parent_node)) # get id for folder from report portal
-              # get child id from other process
+            is_created = false
+            if parallel? && name.include?("Folder:")
+              folder_name = name.gsub("Folder: ", "")
+              folder_name_for_tracker = "./#{folder_name}" # create a full path file name to use for tracking
+              if index > 0
+                folder_name_for_tracker = "./"
+                for path_index in (0...path_components_no_feature.length)
+                  folder_name_for_tracker += "#{path_components_no_feature[path_index]}/"
+                end
+              end
+              $folder_creation_tracking_file = (Pathname(Dir.tmpdir)) + "folder_creation_tracking_#{ReportPortal.launch_id}.lck"
+              File.open($folder_creation_tracking_file, 'r+') do |f|
+                f.flock(File::LOCK_SH)
+                report_portal_folders = f.read
+                if report_portal_folders
+                  report_portal_folders_array = report_portal_folders.split(/\n/)
+                  if report_portal_folders_array.include?(folder_name_for_tracker)
+                    is_created = true
+                  end
+                end
+                f.flock(File::LOCK_UN)
+              end
+              unless is_created
+                File.open($folder_creation_tracking_file, 'a') do |f|
+                  f.flock(File::LOCK_EX)
+                  f.write("\n#{folder_name_for_tracker}")
+                  f.flush
+                  f.flock(File::LOCK_UN)
+                end
+              end
+            end
+            if parallel? && index < path_components.size - 1 && is_created
+              id_of_created_item = ReportPortal.item_id_of(name, parent_node)
```

Comment:
@DzmitryHumianiuk @aliaksandr-trush This will handle the cases described where there are 2 sub-directories of the same name but the parent directory is different, I am constructing a "full path", my logic is there and I tested it but could use a keen 👁